tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -178,6 +178,221 @@ def test_prod_dtype(self, xp, src_dtype, dst_dtype):
         return a.prod(dtype=dst_dtype)
 
 
+@testing.gpu
+class TestNansumNanprod(unittest.TestCase):
```

Comment:
How about parameterizing some parameters (e.g., shapes, axis, Ellipsis...) with `cupy.testing.parameterize`?

Reply:
I did some parameterization now, but it's still quite long, in some cases because we can't parameterize the decorators (at least I haven't found a way to do that in CuPy tests).

Reply:
I'm sorry but it seems hard to fully parameterize them. How about writing as follows for example for code simplification?
```py
def _nan_shaped_arange(shape, xp, dtype, transpose_axes=None):
    a = testing.shaped_arange(shape, xp, dtype)
    if transpose_axes is not None:
        a = a.transpose(transpose_axes)
    if not issubclass(dtype, xp.integer):
        a[:, 1] = xp.nan

@testing.for_all_dtypes(no_bool=True)
@testing.numpy_cupy_allclose()
def test_nansum_all(self, xp, dtype):
    a = _nan_shaped_arange(self.shape, xp, dtype)
    return xp.nansum(a)
```

Reply:
I'm always up to simplify code and avoid duplication. I intentionally did not want to do this before because I haven't seen many such tests before in CuPy, so I thought you had a reason to be mostly explicit. I did now some more parameterization, let me know what you think.

In particular, let me point out to https://github.com/cupy/cupy/pull/2252/commits/ee2793ae4db775fbf38f0d0e0327c96f7f2df3d4#diff-a8120504756ee4e24b8b06b606374e65R210, for that I added two tests that are almost the same, with the exception of adding `@testing.numpy_cupy_allclose(contiguous_check=False)` for one of them and just not having the test do anything (`return xp.array(())`) if the appropriate condition isn't met. This isn't the cleanest, but again we fall into the case of not being able to parameterize the decorators, so hopefully that's an acceptable workaround.